examples/Reduced_normal_equations.ipynb

```diff
@@ -0,0 +1,633 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/jmapel/miniconda3/envs/knoten/lib/python3.7/site-packages/ale/__init__.py:22: UserWarning: ALESPICEROOT environment variable not set, Spice Drivers will not function correctly\n",
+      "  warnings.warn('ALESPICEROOT environment variable not set, Spice Drivers will not function correctly')\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "from plio.io import io_controlnetwork\n",
+    "from knoten.csm import create_csm\n",
+    "from scipy import sparse\n",
+    "import ale\n",
+    "import csmapi\n",
+    "import numpy as np\n",
+    "from collections import OrderedDict\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "from knoten.bundle import *"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def compute_image_weight(sensor, parameters):\n",
+    "    \"\"\"\n",
+    "    Compute the weight matrix for the sensor parameters of a single image.\n",
+    "    \n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    sensor : csmapi.RasterGm\n",
+    "             The CSM sensor model for the image\n",
+    "    parameters : list\n",
+    "                 The list of parameters to solve for\n",
+    "    \"\"\"\n",
+    "    param_count = len(parameters)\n",
+    "    covar_mat = np.zeros((param_count, param_count))\n",
+    "    for a, b in itertools.product(range(param_count), range(param_count)):\n",
+    "        covar_mat[a, b] = sensors[sn].getParameterCovariance(parameters[a].index, parameters[b].index)\n",
+    "    return np.linalg.inv(covar_mat)\n",
+    "\n",
+    "def compute_point_weight(cnet, point_id):\n",
+    "    \"\"\"\n",
+    "    Compute the weight matrix for a single point.\n",
+    "    \n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    cnet : DataFrame\n",
+    "           The control network dataframe\n",
+    "    point_id : str\n",
+    "               The point ID string of the point to compute the weight matrix for\n",
+    "    \"\"\"\n",
+    "    covar_vec = cnet.loc[(cnet['id'] == point_id).idxmax(), 'aprioriCovar']\n",
+    "    if hasattr(covar_vec, '__len__') and len(covar_vec) == 6:\n",
+    "        return np.linalg.inv(np.array(\n",
+    "            [[covar_vec[0], covar_vec[1], covar_vec[2]],\n",
+    "             [covar_vec[1], covar_vec[3], covar_vec[4]],\n",
+    "             [covar_vec[2], covar_vec[4], covar_vec[5]]]))\n",
+    "\n",
+    "    return np.zeros((3,3))\n",
+    "\n",
+    "def update_parameters(sensors, parameters, network, updates, coefficient_columns):\n",
```

Comment:
I am confused why we haven't put these functions into bundle.py yet. Just avoiding test writing during this developmental/experimental phase?

Reply:
laziness really. I'll work on getting these integrated into bundle.py.

Reply:
👍 😆